packages/donor/src/components/basic/IconButton/IconButton.tsx

```diff
@@ -3,23 +3,27 @@ import {
   IconButtonProps as MuiIconButtonProps,
 } from "@mui/material";
 import { AnalyticsButtonType } from "@zm-blood-components/common";
+import { MouseEvent } from "react";
 import { reportClick } from "../../../Analytics";
 
 export interface IconButtonProps extends MuiIconButtonProps {
   /** For logging and Analytics */
   buttonName: string;
-  onClick: () => void;
 }
 
-export default function IconButton({
-  buttonName,
-  onClick,
-  ...props
-}: IconButtonProps) {
-  const handleClick = () => {
-    onClick();
+export default function IconButton({ buttonName, ...props }: IconButtonProps) {
+  const handleClick = (
+    e: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
+  ) => {
+    props.onClick?.(e);
```

Comment:
Please explain this change

Reply:
This was unnecessary.
I undid the change and converted UpcomingDonationScreen Line #44 to work with React Ref.